src/components/FileItem.tsx

```diff
@@ -27,11 +28,18 @@ export const STATUS_CODES = {
 
 export interface IFileItemProps {
   actions?: React.ReactElement;
-  contextMenu: (event: React.MouseEvent) => void;
+  contextMenu?: (event: React.MouseEvent) => void;
   file: Git.IStatusFile;
+  markBox?: boolean;
   model: GitExtension;
-  selected: boolean;
-  selectFile: (file: Git.IStatusFile | null) => void;
+  selected?: boolean;
+  selectFile?: (file: Git.IStatusFile | null) => void;
```

Comment:
Would it be better to set an function doing nothing rather than making `contextMenu` and `selectFile` optional?

Reply:
My current mental model of FileItem is as an object with configurable support for optional features. For example, in the Simple Staging case it may never make sense for a FileItem to have selection support. Thus, it makes sense for eg `selectFile` to be optional, rather than require a dummy